### PR TITLE
Fix FIFO for userscripts on Windows

### PR DESCRIPTION
**Pull request by Kingdread**
Due to the way Windows handles files, we couldn't actually write to the "FIFO" file from the userscript (same issue as #1767). This patch should fix this problem with the patch [from the mailinglist](https://lists.schokokeks.org/pipermail/qutebrowser/2016-September/000256.html), and it adds a simple test for running a userscript on Windows.

On a somewhat related note: I'm wondering how those tests even pass on AppVeyor - `echo` is not an external command on Windows (there is no `echo.exe`), and you can't call `echo` via `subprocess`. The tests fail locally, but seem to work fine on AppVeyor... Any idea what magic you used to get it to work? :wink:

### CODE REVIEW

File: qutebrowser/commands/userscripts.py [hunk 1]
```diff
@@ -280,14 +280,10 @@ class _WindowsUserscriptRunner(_BaseUserscriptRunner):
...
     This also means the userscript *has* to use >> (append) rather than >
     (overwrite) to write to the file!
-
-    Attributes:
-        _oshandle: The oshandle of the temp file.
     """
 
     def __init__(self, win_id, parent=None):
         super().__init__(win_id, parent)
-        self._oshandle = None
```
Comment (The-Compiler) [inline qutebrowser/commands/userscripts.py]: You can remove the overridden `__init__`now.

File: tests/end2end/data/userscripts/open_current_url.bat [hunk 2]
```diff
@@ -0,0 +1 @@
+echo open -t %QUTE_URL% >> "%QUTE_FIFO%"
```
Comment (The-Compiler) [inline tests/end2end/data/userscripts/open_current_url.bat]: nitpick, please add a final newline (but no blank line) at the end of the file :wink:

File: tests/end2end/features/spawn.feature [hunk 3]
```diff
@@ -40,3 +40,11 @@ Feature: :spawn
         Then the following tabs should be open:
             - about:blank
             - about:blank (active)
+    @windows
```
Comment (The-Compiler) [inline tests/end2end/features/spawn.feature]: And another nitpick, please add a blank line before this

**Comment by The-Compiler** [comment]
Seems like some freshly added tests are flaky, I'll take a look.

File: tests/end2end/data/userscripts/open_current_url.bat (hunk 2 above)
Comment (Kingdread) [inline tests/end2end/data/userscripts/open_current_url.bat]: Seems like I should finally set up an editor on Windows correctly if I continue to do this Windows stuff :laughing:

**Comment by The-Compiler** [comment]
After some fixes and [librally skipping everything broken](https://github.com/The-Compiler/qutebrowser/commit/08302c5a5fdc63e608c628b838a926d8c66e30dc) for now, things are back to normal again :laughing:

**Comment by The-Compiler** [comment]
> On a somewhat related note: I'm wondering how those tests even pass on AppVeyor - `echo` is not an external command on Windows (there is no `echo.exe`), and you can't call `echo` via `subprocess`. The tests fail locally, but seem to work fine on AppVeyor... Any idea what magic you used to get it to work? :wink:

Right, they are broken here too... My best guess would be that AppVeyor has some cygwin `echo.exe` in `%PATH%`? :laughing: 

I wonder what the best way to fix them would be - is there an executable which is there and does nothing on Windows/Linux/OS X? Or do we need a `(python)` replacement there which expands to `sys.executable` and work with that?

**Comment by Kingdread** [comment]
> My best guess would be that AppVeyor has some cygwin echo.exe in %PATH%?

Seems quite possible, so I'll accept that as an explanation :stuck_out_tongue_winking_eye: 
```
C:\Users\Daniel>C:\cygwin64\bin\echo.exe "foobar"
foobar
```

> is there an executable which is there and does nothing on Windows/Linux/OS X?

One for all platforms? Probably not. But if the name can vary, it should be easy to find something for each one separately. For Windows, see [this question](http://superuser.com/questions/381103/is-there-a-windows-exe-that-does-nothing), or we could drop a simple `.bat` into the userscript dir.

Or we use some `(python)` thing, as you suggested, as we're going to need a platform-dependent variable anyway.

**Comment by Kingdread** [comment]
> FAIL tests/end2end/features/test_tabs_bdd.py::test_buffer_with_a_matching_title

Seems unrelated to my changes and fixed in `master`, so I've restarted the Travis build to see if everything passes now.

**Comment by The-Compiler** [comment]
Thanks!

**Pull request merged**